### PR TITLE
Update spectral match results visualizer

### PR DESCRIPTION
**Pull request by robinschmid**
Hey Tomas,

this PR is an update for the spectral matching results window. 
Ansgar has done a great job on the metadata panel. And I have changed the data model and the charts. Now every matching step is visualized in a different color:
 - filtered (black): Multiple options
    - 13C isotope filter (see image below)
    - crop to overlapping m/z range (first image, left side is black, because m/z range was different - or fragmentation energy was differnt)
 - aligned (green): Aligned signals in query and library spectrum (within m/z tolerance)
 - unaligned (orange)

image
image

### CODE REVIEW

File: src/main/java/net/sf/mzmine/modules/visualization/spectra/simplespectra/spectraidentification/spectraldatabase/SpectraIdentificationResultsWindow.java [hunk 1]
```diff
@@ -406,12 +430,15 @@ else if (smilesString != "N/A") {
...
     panelOther.setBackground(Color.WHITE);
     JLabel otherInfo = new JLabel("Other information");
     otherInfo.setToolTipText("This section shows all other information of the matched compound");
     otherInfo.setFont(headerFont);
-    panelOther.add(otherInfo);
+    constraintsMetaDataPanel.gridx = 1;
+    constraintsMetaDataPanel.gridy = 3;
+    metaDataPanel.add(otherInfo, constraintsMetaDataPanel);
     SpectralIdentificationSpectralDatabaseTextPane investigator =
         new SpectralIdentificationSpectralDatabaseTextPane("Prinziple investigator: "
```
Comment (tomas-pluskal) [inline src/main/java/net/sf/mzmine/modules/visualization/spectra/simplespectra/spectraidentification/spectraldatabase/SpectraIdentificationResultsWindow.java]: prinziple -> principal

File: src/main/java/net/sf/mzmine/modules/visualization/spectra/simplespectra/mirrorspectra/MirrorScanWindow.java [hunk 2]
```diff
@@ -103,24 +112,96 @@ public void setScans(PeakListRow row, SpectralDBPeakIdentity db) {
...
+    // create without data
+    mirrorSpecrumPlot = SpectrumChartFactory.createMirrorChartPanel(
+        "Query: " + scan.getScanDefinition(), precursorMZA, rtA, null, "Library: " + db.getName(),
+        precursorMZB == null ? 0 : precursorMZB, rtB, null, false, true);
+    mirrorSpecrumPlot.setMaximumDrawWidth(4200);
+    mirrorSpecrumPlot.setMaximumDrawHeight(2500);
+    // add data
+    DataPointsTag[] tags =
+        new DataPointsTag[] {DataPointsTag.ORIGINAL, DataPointsTag.FILTERED, DataPointsTag.ALIGNED};
+    Color[] colors = new Color[] {Color.black, new Color(0xF57C00), new Color(0x388E3C)};
```
Comment (tomas-pluskal) [inline src/main/java/net/sf/mzmine/modules/visualization/spectra/simplespectra/mirrorspectra/MirrorScanWindow.java]: This should definitely be defined as a public static final constant, with comments on what the hex colors are.

File: src/main/java/net/sf/mzmine/util/maths/similarity/spectra/SpectraSimilarity.java [hunk 3]
```diff
@@ -18,16 +18,52 @@
...
 
+import java.util.Arrays;
+import java.util.List;
+import net.sf.mzmine.datamodel.DataPoint;
+import net.sf.mzmine.util.DataPointSorter;
+import net.sf.mzmine.util.SortingDirection;
+import net.sf.mzmine.util.SortingProperty;
+import net.sf.mzmine.util.scans.ScanAlignment;
+
 public class SpectraSimilarity {
```
Comment (tomas-pluskal) [inline src/main/java/net/sf/mzmine/util/maths/similarity/spectra/SpectraSimilarity.java]: Could you add a brief javadoc to explain the purpose of this class?

**Comment by robinschmid** [comment]
I am going to finish this later today. I was away this weekend at the edersee lake.

**Comment by robinschmid** [comment]
- Javadoc was added
- Moved SpectralSimilarity from util/maths/similarity/spectra  to util/scan/similarity
- Added an extra module to sort SpectralDBPeakIdentities of the search module
- correct spelling

**Comment by robinschmid** [comment]
It is all ready for merging from my side.

**Comment by Annexhc** [comment]
Can you explain why you have added a new module for the sorting?

**Comment by robinschmid** [comment]
Modularity reasons, if someone searches multiple data bases, filters the results or changes the sorting manually - this module gives the options to ...
... sort
... filter

**Comment by Annexhc** [comment]
Ah okay, good point that you may use different database for a single analysis. Good addition!

**Comment by Annexhc** [comment]
From my side every is ready for merging as well.

**Comment by tomas-pluskal** [comment]
Looks good!

**Pull request merged**